sql/analyzer/replace_pk_sort.go

```diff
@@ -16,57 +20,72 @@ func replacePkSort(ctx *sql.Context, a *Analyzer, n sql.Node, scope *plan.Scope,
 func replacePkSortHelper(ctx *sql.Context, scope *plan.Scope, node sql.Node, sortNode *plan.Sort) (sql.Node, transform.TreeIdentity, error) {
 	switch n := node.(type) {
 	case *plan.Sort:
-		sortNode = n // TODO: this only preserves the most recent Sort node
-	// TODO: make this work with IndexedTableAccess, if we are statically sorting by the same col
-	case *plan.ResolvedTable:
-		// No sort node above this, so do nothing
-		if sortNode == nil {
+		sortNode = n // lowest parent sort node
+	case *plan.IndexedTableAccess:
```

Comment:
should we have some tests for this new behavior outside of min/max? Or does this only apply to min/max?

Reply:
This also impacts sorting over primary keys with a filter that has been pushed down, will add plan tests for this.